sklearn/gp_search.py

```diff
@@ -0,0 +1,421 @@
+"""
+The :mod:`sklearn.gp_search` includes utilities to fine-tune the parameters
+of an estimator through a Gaussian Process model.
+"""
+from __future__ import print_function
+
+# Author: Sebastien Dubois <http://bit.ly/SebastienDubois>,
+# License: BSD 3 clause
+
+import numpy as np
+from scipy.stats import norm
+
+from .gaussian_process import GaussianProcessRegressor
+from .cross_validation import check_cv
+from .cross_validation import _fit_and_score
+from .metrics.scorer import check_scoring
+from .base import is_classifier, clone
+from .grid_search import BaseSearchCV
+
+
+def _sample_candidates(n_candidates, param_bounds, param_is_int):
+    n_parameters = param_is_int.shape[0]
+    candidates = []
+
+    for k in range(n_parameters):
+        if param_is_int[k]:
+            k_sample = np.asarray(
+                np.random.rand(n_candidates)
+                * np.float(param_bounds[k][1]-param_bounds[k][0])
+                + param_bounds[k][0],
+                dtype=np.int32)
+        else:
+            k_sample = np.asarray(
+                np.random.rand(n_candidates)
+                * np.float(param_bounds[k][1]-param_bounds[k][0])
+                + param_bounds[k][0])
+        candidates.append(k_sample)
+
+    candidates = np.asarray(candidates).T
+    return unique_rows(candidates)
+
+
+def compute_expected_improvement(predictions, sigma, y_best):
+    ei_array = np.zeros(predictions.shape[0])
+    for i in range(ei_array.shape[0]):
+        z = (y_best - predictions[i]) / sigma[i]
```

Comment:
Shouldnt it be `(predictions[i] - y_best) / sigma[i]` instead? (see equation 4 of http://arxiv.org/pdf/1012.2599v1)